### PR TITLE
Mss 911 fix notifications kicker

### PR DESCRIPTION
**Pull request by NathanielBennett**
Fixes the kicker bug
Makes the title of notification an optional string. 
Needs to be released at the same time as https://github.com/guardian/mobile-n10n/pull/464

### CODE REVIEW

File: src/main/scala/com/gu/mobile/content/notifications/lib/ContentAlertPayloadBuilder.scala [hunk 1]
```diff
@@ -82,16 +82,12 @@ trait ContentAlertPayloadBuilder extends Logging {
...
   }
 
-  private def contentTitle(content: Content, followableTag: Option[Tag], topics: List[Topic]): String = {
-    def title = content.fields.flatMap { cf => cf.headline }.getOrElse(content.webTitle)
-    def reason = followableTag.map { ft => ft.webTitle }.getOrElse("Following")
-
+  private def contentTitle(content: Content, followableTag: Option[Tag], topics: List[Topic]): String =
     if (topics.toSet.intersect(topicsWithoutPrefix).nonEmpty) {
-      title
+      ""
```
Comment (alexduf) [inline src/main/scala/com/gu/mobile/content/notifications/lib/ContentAlertPayloadBuilder.scala]: It looks like you made the title optional but you're still returning an empty string here. Should that be returning None instead?

File: src/test/scala/com/gu/mobile/content/notifications/lib/ContentAlertPayloadBuilderSpec.scala [hunk 2]
```diff
@@ -138,23 +132,23 @@ class ContentAlertPayloadBuilderSpec extends MockitoSugar with WordSpecLike with
       val tag = Tag("membership/series/weekend-round-up", TagType.Series, None, None, "Steve", "", "")
       val topic = Topic(TagSeries, "membership/series/weekend-round-up")
       val minuteItem = item.copy(tags = List(tag))
-      val expectedMinutePayload = expectedPayloadForItem.copy(title = "headline", topic = List(topic))
+      val expectedMinutePayload = expectedPayloadForItem.copy(title = Some(""), topic = List(topic))
```
Comment (alexduf) [inline src/test/scala/com/gu/mobile/content/notifications/lib/ContentAlertPayloadBuilderSpec.scala]:
```suggestion
      val expectedMinutePayload = expectedPayloadForItem.copy(title = None, topic = List(topic))
```

File: src/test/scala/com/gu/mobile/content/notifications/lib/ContentAlertPayloadBuilderSpec.scala [hunk 3]
```diff
@@ -138,23 +132,23 @@ class ContentAlertPayloadBuilderSpec extends MockitoSugar with WordSpecLike with
...
       builder.buildPayLoad(minuteItem) mustEqual expectedMinutePayload
     }
 
     "do not prefix title for weekend reading" in {
       val tag = Tag("membership/series/weekend-reading", TagType.Series, None, None, "Steve", "", "")
       val topic = Topic(TagSeries, "membership/series/weekend-reading")
       val minuteItem = item.copy(tags = List(tag))
-      val expectedMinutePayload = expectedPayloadForItem.copy(title = "headline", topic = List(topic))
+      val expectedMinutePayload = expectedPayloadForItem.copy(title = Some(""), topic = List(topic))
```
Comment (alexduf) [inline src/test/scala/com/gu/mobile/content/notifications/lib/ContentAlertPayloadBuilderSpec.scala]:
```suggestion
      val expectedMinutePayload = expectedPayloadForItem.copy(title = None, topic = List(topic))
```

File: src/test/scala/com/gu/mobile/content/notifications/lib/ContentAlertPayloadBuilderSpec.scala [hunk 4]
```diff
@@ -138,23 +132,23 @@ class ContentAlertPayloadBuilderSpec extends MockitoSugar with WordSpecLike with
...
       builder.buildPayLoad(minuteItem) mustEqual expectedMinutePayload
     }
 
     "do not prefix title for guardian-morning-briefing" in {
       val tag = Tag("world/series/guardian-morning-briefing", TagType.Series, None, None, "Steve", "", "")
       val topic = Topic(TagSeries, "world/series/guardian-morning-briefing")
       val contentItem = item.copy(tags = List(tag))
-      val expectedPayload = expectedPayloadForItem.copy(title = "headline", topic = List(topic))
+      val expectedPayload = expectedPayloadForItem.copy(title = Some(""), topic = List(topic))
```
Comment (alexduf) [inline src/test/scala/com/gu/mobile/content/notifications/lib/ContentAlertPayloadBuilderSpec.scala]:
```suggestion
      val expectedPayload = expectedPayloadForItem.copy(title = None, topic = List(topic))
```

File: src/test/scala/com/gu/mobile/content/notifications/lib/ContentAlertPayloadBuilderSpec.scala [hunk 5]
```diff
@@ -163,7 +157,7 @@ class ContentAlertPayloadBuilderSpec extends MockitoSugar with WordSpecLike with
       val briefingTopic = Topic(TagSeries, "us-news/series/the-campaign-minute-2016")
       val expectedTitle = "headline"
       val contentItem = item.copy(tags = List(briefingTag))
-      val expectedMinutePayload = expectedPayloadForItem.copy(title = expectedTitle, topic = List(briefingTopic))
+      val expectedMinutePayload = expectedPayloadForItem.copy(title = Some(""), topic = List(briefingTopic))
```
Comment (alexduf) [inline src/test/scala/com/gu/mobile/content/notifications/lib/ContentAlertPayloadBuilderSpec.scala]:
```suggestion
      val expectedMinutePayload = expectedPayloadForItem.copy(title = None, topic = List(briefingTopic))
```

**Comment by alexduf** [comment]
If I may suggest the release procedure:

1. [x] Merge the n10n PR, let it build but block deployment
1. [x] Throttle the mobile-notification-content lambda in order to retain all outgoing notifications
1. [x] Deploy the n10n PR
1. [ ] Deploy mobile-notification-content lambda
1. [ ] Un-throttle the lambda

_in theory_ that means no notification is sent during deployment, and the lambda should catch up by processing any missed event on the kinesis stream.

**Comment by NathanielBennett** [comment]
Yes that's a good plan. I'll do it on Monday Morning, though

**Review by alexduf — CHANGES_REQUESTED** [review]

**Review by alexduf — APPROVED** [review]
I think it looks good

**Pull request merged**